lib/mixpanel/client.ex

```diff
@@ -35,53 +45,140 @@ defmodule Mixpanel.Client do
     GenServer.cast(process || __MODULE__, {:engage, event})
   end
 
-  def init({:ok, config}) do
-    {:ok, Enum.into(config, %{})}
+  def init({:ok, opts}) do
+    config = Enum.into(opts, @defaults)
+
+    queues = %{
+      track: Queue.new(),
+      engage: Queue.new()
+    }
+
+    {:ok, {config, queues}}
   end
 
   # No events submitted when env configuration is set to false.
-  def handle_cast(_request, %{active: false} = state) do
+  def handle_cast(_request, {%{active: false}, _} = state) do
     {:noreply, state}
   end
 
-  def handle_cast({:track, event, properties}, %{token: token} = state) do
-    data =
-      %{event: event, properties: Map.put(properties, :token, token)}
-      |> Poison.encode!()
-      |> :base64.encode()
+  def handle_cast({:track, _event, _properties} = event, {config, state}) do
+    case Queue.push(state.track, event, config.max_queue_track) do
+      :dropped ->
+        :telemetry.execute([:mixpanel, :dropped, :track], %{count: 1})
```

Comment:
Hmm, maybe I should instead increase a counter in the state instead of calling `:telemetry` in a hot loop. I can then report the total number dropped when the process is idle

Reply:
Depends on what happens in Telemetry handler. 
If it is just call that increments StatsD counter, it is really cheap: one udp call over localhost.